frontend/src/pages/NewRun.tsx

```diff
@@ -798,6 +798,13 @@ class NewRun extends Page<{}, NewRunState> {
       pipeline_spec: {
         parameters: (this.state.parameters || []).map(p => {
           p.value = (p.value || '').trim();
+          // Remove any special formatting for displaying jsons on an editor
+          try {
+            const parsedJson = JSON.stringify(JSON.parse(p.value || ''));
```

Comment:
Is this more like `formattedJson`? I'm curious why would this be useful. Can you also add a test case for this in NewRun.test.tsx?

Reply:
Yes, I think `unformattedJson` would be best. If you leave the json editor open and click start run then p.value will include all of the newline and whitespace characters used to display the json in the editor. This wouldn't be an issue for most standardized json parsers, but could be an issue for a custom parser. 

Another option would be to add `formattedJson` in the state for ParamEditor which holds the json string with any formatting done in `getDerivedStateFromProps` which would remove the need for any checks in NewRun.tsx. But this would mean an extra parse and stringify would be needed onChange and there would be a some redundancy between props and state. I prefer the original solution, but I will defer to you.

Reply:
I don't agree on this. Here's my reasoning:

* With the auto formatting at pipeline submit time, a user doesn't know the parameter is formatted before it is sent out. If there's something unexpected happens that don't work with formatted result. It would be really hard to identify the root cause.
* Without auto formatting, a user knows the parameter is exactly the same as in his editor on the UI. If something wrong happens, the user will know they could either close the editor to restore it to one line. Or fix their own system that doesn't handle standard JSON.

In general, I think KFP UI shouldn't secretly help user preprocess parameter. What the user sees on the UI should be the value we send to APIs.

Reply:
Okay that makes sense, I removed it from NewRun.tsx